lib/matplotlib/cbook.py

```diff
@@ -2661,3 +2661,203 @@ def __exit__(self, exc_type, exc_value, traceback):
+                                        lambda p: p[0] > 0)
+
+    def __init__(self, str_func):
+        """
+        Parameters
+        ----------
+        str_func : string
+            String to be parsed.
+
+        """
+        try:  # For python 2.7 and python 3+ compatibility
+            is_str = isinstance(str_func, basestring)
+        except NameError:
+            is_str = isinstance(str_func, str)
+
+        if not is_str:
+            raise ValueError("The argument passed is not a string.")
+        self._str_func = str_func
+        self._key, self._params = self._get_key_params()
+        self._func = self.func
+
+    @property
+    def func(self):
+        """
+        Returns the _FuncInfo object, replacing the relevant parameters if
+        necessary in the lambda functions.
+
+        """
+
+        func = self._funcs[self._key]
+        if self._params:
+            m = func.direct
+            direct = (lambda x, m=m: m(x, self._params))
+
+            m = func.inverse
+            inverse = (lambda x, m=m: m(x, self._params))
+
+            is_bounded_0_1 = func.is_bounded_0_1(self._params)
+
+            func = _FuncInfo(direct, inverse,
+                             is_bounded_0_1)
+        else:
+            func = _FuncInfo(func.direct, func.inverse,
+                             func.is_bounded_0_1())
+        return func
+
+    @property
+    def directfunc(self):
+        """
+        Returns the callable for the direct function.
+
+        """
+        return self._func.direct
+
+    @property
+    def invfunc(self):
+        """
+        Returns the callable for the inverse function.
+
+        """
+        return self._func.inverse
+
+    @property
+    def is_bounded_0_1(self):
+        """
+        Returns a boolean indicating if the function is bounded
+        in the [0-1 interval].
+
+        """
+        return self._func.is_bounded_0_1()
+
+    def _get_key_params(self):
+        str_func = six.text_type(self._str_func)
+        # Checking if it comes with parameters
+        regex = '\{(.*?)\}'
+        params = re.findall(regex, str_func)
+
+        if params:
+            for i in range(len(params)):
+                try:
+                    params[i] = float(params[i])
+                except:
+                    raise ValueError("Error with parameter number %i: '%s'. "
+                                     "'p' in parametric function strings must "
+                                     " be replaced by a number that is not "
+                                     "zero, e.g. 'log10(x+{0.1})'." %
+                                     (i, params[i]))
+
+            str_func = re.sub(regex, '{p}', str_func)
+
+        try:
+            func = self._funcs[str_func]
+        except:
+            raise ValueError("%s: invalid string. The only strings "
+                             "recognized as functions are %s." %
+                             (str_func, self._funcs.keys()))
```

Comment:
Minor tweak: formatting of the string on python 3 will be better if you replace `self._funcs.keys()` with `[k for k in self._funcs.keys()]`.  This is redundant on python 2, but still works.